trustmanager/keyfilestore.go

```diff
@@ -106,54 +96,79 @@ func (s *KeyMemoryStore) ListKeys() []string {
 	return listKeys(s)
 }
 
-func addKey(s LimitedFileStore, name string, privKey data.PrivateKey) error {
-	pemPrivKey, err := KeyToPEM(privKey)
-	if err != nil {
-		return err
-	}
-
-	return s.Add(name, pemPrivKey)
+// RemoveKey removes the key from the keystore
+func (s *KeyMemoryStore) RemoveKey(name string) error {
+	return remove(s, name)
 }
 
-func getKey(s LimitedFileStore, name string) (data.PrivateKey, error) {
-	keyBytes, err := s.Get(name)
+
+func addKey(s LimitedFileStore, passphraseRetriever PassphraseRetriever, name string, privKey data.PrivateKey) error {
```

Comment:
We discussed the need to cache decrypted keys. That will address the current problem where we'd prompt for a password for each key.

As far as addKey, I feel like it's best to push the passphrase retrieval logic to the caller and pass in a password. That way, if a caller adds several keys, it can call the PassphraseRetriever and explicitly use that passphrase for each of the keys. The alternative of having this logic in the PassphraseRetriever callback means that the libnotary user needs to keep track of a lot of state to do their own password caching, and it would make it more complicated to use libnotary.